### PR TITLE
New font provider - loading bitmap fonts

### PR DESCRIPTION
**Pull request by konopka90**
Based on PR #34 . Should work with latest master.

### CODE REVIEW

**Comment by wheybags** [comment]
So to use this, you just have to create a BMFont style .FNT file, and load it like you normally would a freetype font, correct?
You don't need to implement an interface yourself like you do the the render interface?

**Comment by wheybags** [comment]
Would be good to add a sample of the Arial font you supplied being used

File: Source/Core/BitmapFont/FontFaceLayer.cpp [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+FontFaceLayer::FontFaceLayer() : Rocket::Core::FontFaceLayer()
+{
+    handle = NULL;
+    effect = NULL;
+}
+
+FontFaceLayer::~FontFaceLayer()
+{
+    //if (effect != NULL)
```
Comment (wheybags) [inline Source/Core/BitmapFont/FontFaceLayer.cpp]: A few bits of commented out code around here

File: Source/Core/BitmapFont/FontProvider.cpp [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+        Log::Message(Log::LT_ERROR, "Failed to load font face %s (from %s).", bm_font->Face.FamilyName.CString(), file_name.CString());
+        return false;
+    }
+
+    return true;
+}
+
+bool FontProvider::LoadFontFace(const byte* data, int data_length)
+{
+    //    BM_Font *bm_font = (BM_Font*) instance->LoadFace(data, data_length, family, false);
```
Comment (wheybags) [inline Source/Core/BitmapFont/FontProvider.cpp]: Commented out code again, also looks like these functions aren't necessary?

File: Source/Core/FontFaceHandle.cpp [hunk 3]
```diff
@@ -353,167 +307,22 @@ void FontFaceHandle::OnReferenceDeactivate()
...
-				}
-				break;
-			}
-		}
-	}
-	else
-		glyph.bitmap_data = NULL;
-}
-
 int FontFaceHandle::GetKerning(word lhs, word rhs) const
```
Comment (wheybags) [inline Source/Core/FontFaceHandle.cpp]: again unnecessary function?

**Comment by wheybags** [comment]
Apart from the few things I commented on, looks great to me.
@dwimsey  care to take a look?

**Comment by wheybags** [comment]
BTW, there's an example of this working at https://github.com/konopka90/freeablo/tree/bitmap_font

**Comment by dwimsey** [comment]
I'll take a gander tonight, definitely want to get this sort of thing available

**Comment by konopka90** [comment]
There are still some problems with displaying text with property display:inline and handling click event (or all events , dunno so far), maybe tomorrow i'll look at this.

**Comment by konopka90** [comment]
It's working quite good now. Left some TODO's, maybe in future.

**Comment by wheybags** [comment]
All looks good to me now. Click events work in freeablo anyway.
@dwimsey did you have a chance to look at this?
Any problem with me merging it?

**Pull request merged**